src/hsm/PSFCorr.cpp

```diff
@@ -1881,6 +1881,54 @@ namespace hsm {
         float sky_var, const char* shear_est, const std::string& recompute_flux,
         double guess_sig_gal, double guess_sig_PSF, double precision,
         galsim::Position<double> guess_centroid, boost::shared_ptr<HSMParams> hsmparams);
+    template CppShapeData EstimateShearView(
+        const BaseImage<uint16_t>& gal_image, const BaseImage<int>& PSF_image,
+        const BaseImage<int>& gal_mask_image,
+        float sky_var, const char* shear_est, const std::string& recompute_flux,
+        double guess_sig_gal, double guess_sig_PSF, double precision,
+        galsim::Position<double> guess_centroid, boost::shared_ptr<HSMParams> hsmparams);
```

Comment:
Having all possible combinations does seem too much. I followed your suggestion and made the edit to force the gal and psf images into being floats or doubles so there are simply 4 instantiations.